tests/test_date_utils.py

```diff
@@ -75,4 +76,17 @@ def test_to_datetime(arg, expected):
 def test_to_datetime_360(jday_360, month, day):
     assert \
         to_datetime(num2date(jday_360, 'days since 1999-12-30', '360_day')) == \
-        datetime(2000, month, day)
\ No newline at end of file
+        datetime(2000, month, day)
+
+@mark.parametrize('date, resolution, expected', [
+    (datetime(2000, 2, 1, 13, 35), "yearly", datetime(2000, 1, 1)),
+    (datetime(2000, 2, 1, 13, 35), "monthly", datetime(2000, 2, 1)),
+    (datetime(2000, 2, 1, 13, 35), "seasonal", datetime(1999, 12, 1)),
+    (datetime(2000, 2, 1, 13, 35), "30-minute", datetime(2000, 2, 1, 13, 30)),
+    (datetime(2000, 2, 1, 13, 35), "2-minute", datetime(2000, 2, 1, 13, 34)),
+    (datetime(2000, 2, 1, 13, 35), "6-hourly", datetime(2000, 2, 1, 12)),
+    (datetime(2000, 2, 1, 13, 35), "1-hourly", datetime(2000, 2, 1, 13)),
+    (datetime(2012, 5, 1), "seasonal", datetime(2012, 3, 1))
```

Comment:
Should there be tests for other dates than the first of the month? I'd feel more secure if I could see, e.g., the case

```
(datetime(2000, 2, 7, 13, 35), "monthly", datetime(2000, 2, 1)),
```

Although it's possible that case never actually occurs.

Reply:
Tests are cheap, I'll toss some more in. :)